src/v/config/validators.cc

```diff
@@ -120,6 +120,20 @@ validate_sasl_mechanisms(const std::vector<ss::sstring>& mechanisms) {
     return std::nullopt;
 }
 
+std::optional<ss::sstring>
+validate_http_authn_mechanisms(const std::vector<ss::sstring>& mechanisms) {
+    static const absl::flat_hash_set<std::string_view> supported{"BASIC"};
```

Comment:
consider making this thread local to avoid the initialization locking that the compiler will insert. but if this isn't even in the fast path, then you could also avoid pay the memory cost of keeping this around and build it on demand (ie drop `static`)

Reply:
The list is only ever going to be short, so I've made it a `constexpr std::array` with a linear search. This is drastically simpler.